app/src/vanilla/java/cm/aptoide/pt/app/view/AppViewPresenter.java

```diff
@@ -0,0 +1,619 @@
+import cm.aptoide.pt.app.AppViewAnalytics;
+import cm.aptoide.pt.app.AppViewManager;
+import cm.aptoide.pt.app.DownloadAppViewModel;
+import cm.aptoide.pt.crashreports.CrashReport;
+import cm.aptoide.pt.logger.Logger;
+import cm.aptoide.pt.presenter.Presenter;
+import cm.aptoide.pt.presenter.View;
+import cm.aptoide.pt.share.ShareDialogs;
+import java.util.concurrent.TimeUnit;
+import rx.Completable;
+import rx.Observable;
+import rx.Scheduler;
+import rx.Single;
+import rx.android.schedulers.AndroidSchedulers;
+import rx.exceptions.OnErrorNotImplementedException;
+
+/**
+ * Created by franciscocalado on 08/05/18.
+ */
+
+public class AppViewPresenter implements Presenter {
+  private static final long TIME_BETWEEN_SCROLL = 2 * DateUtils.SECOND_IN_MILLIS;
+  private static final String TAG = AppViewPresenter.class.getSimpleName();
+
+  private final PermissionManager permissionManager;
+  private final PermissionService permissionService;
+  private AppViewView view;
+  private AccountNavigator accountNavigator;
+  private AppViewAnalytics appViewAnalytics;
+  private AppViewNavigator appViewNavigator;
+  private AppViewManager appViewManager;
+  private AptoideAccountManager accountManager;
+  private Scheduler viewScheduler;
+  private CrashReport crashReport;
+  private long appId;
+  private String packageName;
+
+  public AppViewPresenter(AppViewView view, AccountNavigator accountNavigator,
+      AppViewAnalytics appViewAnalytics, AppViewNavigator appViewNavigator,
+      AppViewManager appViewManager, AptoideAccountManager accountManager, Scheduler viewScheduler,
+      CrashReport crashReport, long appId, String packageName, PermissionManager permissionManager,
+      PermissionService permissionService) {
+    this.view = view;
+    this.accountNavigator = accountNavigator;
+    this.appViewAnalytics = appViewAnalytics;
+    this.appViewNavigator = appViewNavigator;
+    this.appViewManager = appViewManager;
+    this.accountManager = accountManager;
+    this.viewScheduler = viewScheduler;
+    this.crashReport = crashReport;
+    this.appId = appId;
+    this.packageName = packageName;
+    this.permissionManager = permissionManager;
+    this.permissionService = permissionService;
+  }
+
+  @Override public void present() {
+    handleFirstLoad();
+    handleReviewAutoScroll();
+    handleClickOnScreenshot();
+    handleClickOnVideo();
+    handleClickOnDescriptionReadMore();
+    handleClickOnDeveloperWebsite();
+    handleClickOnDeveloperEmail();
+    handleClickOnDeveloperPrivacy();
+    handleClickOnDeveloperPermissions();
+    handleClickOnStoreLayout();
+    handleClickOnFollowStore();
+    handleClickOnOtherVersions();
+    handleClickOnTrustedBadge();
+    handleClickOnRateApp();
+    handleClickReadComments();
+    handleClickFlags();
+    handleClickLoginSnack();
+    handleClickOnSimilarApps();
+    handleClickOnToolbar();
+    handleDefaultShare();
+    handleRecommendsShare();
+    handleClickOnRetry();
```

Comment:
If every method has handle and click on it, it decreases readability, we should probably discuss this as a team and decide a convention.